### PR TITLE
Automated tests for authentication

### LINKED ISSUE #7: Automated Tests for Authentication
- [ ] Write tests for user registration
- [ ] Write tests for user login

### PR DESCRIPTION
**Pull request by ZaidNayeb**
### Middleware Validation Tests (authMiddleware.test.js)
- [ ] Email validation (required, format)
- [ ] Password validation (required, length, alphanumeric)
- [ ] Login input validation
- [ ] Empty request body handling

 ###Controller Logic Tests (authController.test.js)
- [ ] User registration (success & duplicate email)
- [ ] User login (success, invalid credentials, non-existent user)
- [ ] Password security (hashing, not exposed in responses)
- [ ] JWT token generation
- [ ] Database error handling
- [ ] Edge cases (special characters, long emails)

closes #7

### CODE REVIEW

File: backend/src/_tests_/authController.test.js [hunk 1]
```diff
@@ -151,4 +151,208 @@ describe('Auth Controller - unit tests', () => {
...
+      const req = { body: { email: longEmail, password: 'password123' } };
+      const res = { status: vi.fn().mockReturnThis(), json: vi.fn() };
+
+      UserModel.exists.mockResolvedValue(false);
+      bcrypt.hash.mockResolvedValue('hashed');
+      UserModel.create.mockResolvedValue({ _id: 'user123', email: longEmail });
+
+      await expect(registerUser(req, res)).resolves.not.toThrow();
+    });
+  });
```
Comment (SakinaEbrahimi) [inline backend/src/_tests_/authController.test.js]: also test username and googleId

File: backend/src/_tests_/authMiddleware.test.js [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+
+      const res = {};
+      const next = vi.fn();
+
+      validationResult.mockReturnValue({
+        isEmpty: () => true,
+        array: () => [],
+      });
+
+      await validateFunction(req, res, next);
```
Comment (Mahdi-Hasanzadeh) [inline backend/src/_tests_/authMiddleware.test.js]: No need to use await in here, because all  `validateFunction` is synchronous, you can safely remove await.

**Review by Mahdi-Hasanzadeh — COMMENTED** [review]
Good job.  Just a few small improvements.
1: Your branch is not synced with main branch.
2: In `PR` description try to use checkboxes. - [ ] for what you have done.
3: In `PR` description create a `Close` section to show which issues is going to be closed by this PR.  `Closes #issueNumber`
